data/arrow_test.go

```diff
@@ -293,8 +293,19 @@ func TestEncode(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Check for the same exact file after encode
 	if !bytes.Equal(b, want) {
-		t.Fatalf("data frame doesn't match golden file")
+		// check if the file still represents the same frame or not
```

Comment:
The metadata headers in the golden file are now different, but it continues to read the old files fine.  I added this check to verify that